packages/toolpad-app/src/appDomMigrations/v1.ts

```diff
@@ -0,0 +1,87 @@
+import invariant from 'invariant';
+import { AppDom, QueryNode, createConst, isQuery, ElementNode, isElement, ref } from '../appDom';
+import { update } from '../utils/immutability';
+
+function migrateLegacyQueryNode(node: QueryNode<any>): QueryNode<any> {
+  if (node.attributes.dataSource?.value !== 'rest') {
+    return node;
+  }
+
+  // Migrate old rest nodes with transforms
+  if (
+    typeof node.attributes.query.value?.transformEnabled !== 'undefined' &&
+    (node.attributes.transformEnabled || node.attributes.transform)
+  ) {
+    return update(node, {
+      attributes: update(node.attributes, {
+        transformEnabled: undefined,
+        transform: undefined,
+      }),
+    });
+  }
+
+  if (node.attributes.transformEnabled || node.attributes.transform) {
+    return update(node, {
+      attributes: update(node.attributes, {
+        transformEnabled: undefined,
+        transform: undefined,
+        query: createConst({
+          ...node.attributes.query.value,
+          transformEnabled: node.attributes.transformEnabled?.value,
+          transform: node.attributes.transform?.value,
+        }),
+      }),
+    });
+  }
+
+  // Update all query connectionId
+  if (node.attributes.query.value?.connectionId) {
+    if (typeof node.attributes.query.value?.connectionId?.value === 'string') {
+      return update(node, {
+        attributes: update(node.attributes, {
+          query: createConst({
+            ...node.attributes.query.value,
+            connectionId: createConst(ref(node.attributes.query.value.connectionId.value)),
+          }),
+        }),
+      });
+    }
+  }
+  return node;
+}
+
+function migrateLegacyNavigationAction(node: ElementNode<any>): ElementNode<any> {
```

Comment:
Should we remove https://github.com/mui/mui-toolpad/blob/master/packages/toolpad-app/src/appDom.ts#L903 and https://github.com/mui/mui-toolpad/blob/master/packages/toolpad-app/src/appDom.ts#L890 as well?

Reply:
these need to be moved in a migration

Reply:
Yep, have added these to the first migration in this PR